pkg/plugins/plugincontext/plugincontext.go

```diff
@@ -0,0 +1,104 @@
+package plugincontext
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/grafana/grafana/pkg/plugins"
+
+	"github.com/grafana/grafana/pkg/bus"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/plugins/adapters"
+	"github.com/grafana/grafana/pkg/util/errutil"
+)
+
+type cacheService interface {
+	Get(k string) (interface{}, bool)
+	Set(k string, x interface{}, d time.Duration)
+}
+
+type busDispatcher interface {
+	Dispatch(msg bus.Msg) error
+}
+
+type datasourceGetter interface {
+	GetDatasourceByUID(datasourceUID string, user *models.SignedInUser, skipCache bool) (*models.DataSource, error)
+}
+
+func Get(pluginID string, datasourceUID string, pluginManager plugins.Manager, user *models.SignedInUser, cacheService cacheService, busDispatcher busDispatcher, datasourceGetter datasourceGetter) (backend.PluginContext, bool, error) {
```

Comment:
Nit. This feels a bit awkward having all these dependencies as arguments. Feels like this should be exposed as a service itself (e.g. PluginContextProvider?) that live can depend upon rather than 3 different services. 

Not really a blocker, but something to consider after this has been merged.

Reply:
Moved to separate service. It would be cool to write some tests for it, but maybe leave this as a task for a future since this is mostly layout refactoring here.

Reply:
That's totally fine